lib/sassr.js

```diff
@@ -0,0 +1,164 @@
+        // double-quote marks because templates/sassr.tpl.js wraps the
+        // string in single-quotes anyway.
+        compiled = sassrModuleFn({
+            css: JSON.stringify(compileSync(opts)).slice(1, -1)
+        });
+    } catch (e) {
+        compiled = errorModuleFn({
+            message: e.message
+        });
+    }
+
+    return compiled;
+}
+
+function sassrSync(file, opts) {
+    /* jshint validthis: true */
+    if (!SASS_EXTENSION_RE.test(file)) {
+        return through();
+    }
+
+    var buffers = [];
+
+    function push(chunk, enc, next) {
+        buffers.push(chunk);
+        next();
+    }
+
+    function end(next) {
+        var src = Buffer.concat(buffers).toString();
+        var paths = [
+            path.dirname(file)
+        ].concat(opts.includePaths);
+
+        opts = _.defaults({
+            data: src,
+            includePaths: paths
+        }, opts);
+
+        try {
+            this.push(transformSync(opts));
+        } catch (e) {
+            this.emit('error', e);
+            return;
+        }
+
+        next();
+    }
+
+    return through(push, end);
+}
+
+function compile(opts, cb) {
+    var omitFields = ['file', 'success', 'error'];
+    opts = _.omit(_.defaults({}, opts, SASS_DEFAULTS), omitFields);
+
+    sass.render(opts, cb);
+}
+
+function transform(opts, cb) {
+    if (!opts) { opts = {}; }
+
+    compile(opts, function(error, result) {
+        var module;
+        var errMessage = (error) ? error.message : '';
+
+        if (error) {
+            module = errorModuleFn({
+                message: error.message
+            });
+        } else {
+            // JSON.stringify protects the template function from throwing
+            // `Unexpected token ILLEGAL`. Slicing the leading and trailing
+            // double-quote marks because templates/sassr.tpl.js wraps the
+            // string in single-quotes anyway.
+            module = sassrModuleFn({
+                css: JSON.stringify(result.css.toString()).slice(1, -1)
+            });
+        }
+
+        cb(errMessage, module);
+    });
+}
+
+function sassr(file, opts) {
+    /* jshint validthis: true */
+    if (!SASS_EXTENSION_RE.test(file)) {
+        return through();
+    }
+
+    var buffers = [];
+
+    function push(chunk, enc, next) {
+        buffers.push(chunk);
+        next();
+    }
+
+    function end(next) {
+        var self = this;
+        var src = Buffer.concat(buffers).toString();
+        var paths = [
+            path.dirname(file)
+        ].concat(opts.includePaths);
+
+        opts = _.defaults({
+            data: src,
+            includePaths: paths
+        }, opts);
+
+        transform(opts, function(errMessage, module) {
+            if (errMessage) {
+                self.emit('error', errMessage);
+            }
+            self.push(module);
+            next();
+        });
+    }
+
+    return through(push, end);
+}
+
+module.exports = sassr;
+module.exports.compile = compile;
```

Comment:
i'd probably add the functions onto `sassr` directly instead of `module.exports`